homeassistant/components/enigma2/media_player.py

```diff
@@ -0,0 +1,251 @@
+    STATE_OFF, STATE_ON, STATE_PLAYING, CONF_PORT)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['openwebifpy==1.2.0']
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_MEDIA_CURRENTLY_RECORDING = 'media_currently_recording'
+ATTR_MEDIA_DESCRIPTION = 'media_description'
+ATTR_MEDIA_END_TIME = 'media_end_time'
+ATTR_MEDIA_START_TIME = 'media_start_time'
+
+CONF_PREFER_PICON = "prefer_picon"
+
+DEFAULT_NAME = 'Enigma2 Media Player'
+DEFAULT_PORT = 80
+DEFAULT_SSL = False
+DEFAULT_PREFER_PICON = False
+DEFAULT_USERNAME = 'root'
+DEFAULT_PASSWORD = 'dreambox'
+
+SUPPORTED_ENIGMA2 = SUPPORT_VOLUME_SET | SUPPORT_VOLUME_MUTE | \
+                    SUPPORT_TURN_OFF | SUPPORT_NEXT_TRACK | SUPPORT_STOP | \
+                    SUPPORT_PREVIOUS_TRACK | SUPPORT_VOLUME_STEP | \
+                    SUPPORT_TURN_ON | SUPPORT_PAUSE | SUPPORT_SELECT_SOURCE
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+    vol.Optional(CONF_USERNAME, default=DEFAULT_USERNAME): cv.string,
+    vol.Optional(CONF_PASSWORD, default=DEFAULT_PASSWORD): cv.string,
+    vol.Optional(CONF_SSL, default=DEFAULT_SSL): cv.boolean,
+    vol.Optional(CONF_PREFER_PICON, default=DEFAULT_PREFER_PICON): cv.boolean,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up of an enigma2 media player."""
+    if discovery_info:
+        discovered_config = {
+            CONF_NAME: DEFAULT_NAME,
+            # Do not use the discovered port, as it is the streaming
+            # service port (8001) which is not what we want.
+            CONF_PORT: DEFAULT_PORT,
+            CONF_HOST: discovery_info['host'],
+            CONF_USERNAME: DEFAULT_USERNAME,
+            CONF_PASSWORD: DEFAULT_PASSWORD,
+            CONF_SSL: DEFAULT_SSL,
+            CONF_PREFER_PICON: DEFAULT_PREFER_PICON,
+        }
+        add_devices([Enigma2Device(discovery_info['hostname'],
+                                   discovered_config)], True)
+        return
+
+    add_devices([Enigma2Device(config[CONF_NAME], config)], True)
+
+
+class Enigma2Device(MediaPlayerDevice):
+    """Representation of an Enigma2 box."""
+
+    def __init__(self, name, config):
+        """Initialize the Enigma2 device."""
+        import openwebif.api
+        self._state = None
+        self._name = name
+        self.e2_box = \
+            openwebif.api.CreateDevice(host=config[CONF_HOST],
+                                       port=config[CONF_PORT],
+                                       username=config[CONF_USERNAME],
+                                       password=config[CONF_PASSWORD],
+                                       is_https=config[CONF_SSL],
+                                       prefer_picon=config[CONF_PREFER_PICON])
+
+        self.volume = 10
```

Comment:
Should this be left to `None` and let the update to take care of setting it?

Reply:
Done now in the pypi module.